### PR TITLE
feat(csdx): add generator

### PR DESCRIPTION
**Pull request by davidalpert**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

feature: initial completions for the ContentStack nodejs CLI tool csdx

**Additional info:**

[ContentStack](https://www.contentstack.com/) is a cloud-hosted CMS tool:
> Contentstack is an API-based, headless content management platform that allows developers and content managers to create and manage content simultaneously and independently, to create websites and applications quickly.

It provides a nodejs CLI tool for command-line interaction with your content stacks.
- https://www.contentstack.com/docs/developers/cli/

### CODE REVIEW

File: src/csdx.ts [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+      description:
+        "The auth:login command lets you log in to Contentstack and save the session",
+      options: [
+        {
+          name: ["-u", "--username"],
+          description: "Email address of your Contentstack account",
+          args: {
+            name: "email address",
+          },
+          isOptional: false,
```
Comment (SeparateRecords) [inline src/csdx.ts:16]: For options, the default is to be optional, so the flag is `isRequired`. For optional options, you can omit it entirely!

File: src/csdx.ts [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+      name: "whoami",
+      description:
+        "The csdx auth:whoami command returns the username (email address) of the user who is currently logged in to the session",
+    },
+  ],
+  options: [
+    {
+      name: ["--help", "-h"],
+      description: "Show help for csdx",
+    },
```
Comment (SeparateRecords) [inline src/csdx.ts:356]: Makes --help show on each subcommand
```suggestion
    {
      name: ["--help", "-h"],
      description: "Show help for csdx",
      isPersistent: true,
    },
```

**Review by SeparateRecords — CHANGES_REQUESTED** [review]
Looks good! Main feedback is to flip `isOptional` to `isRequired`, you can remove any `isRequired: false`

File: src/csdx.ts (hunk 1 above)
Comment (davidalpert) [reply src/csdx.ts]: @SeparateRecords thank you for this; fixed in a subsequent commit

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**